napari/layers/base/_base_mouse_bindings.py

```diff
@@ -0,0 +1,179 @@
+import warnings
+
+import numpy as np
+
+from napari.layers.utils.interaction_box import (
+    InteractionBoxHandle,
+    generate_transform_box_from_layer,
+    get_nearby_handle,
+)
+from napari.utils.transforms import Affine
+
+
+def highlight_box_handles(layer, event):
+    """
+    Highlight the hovered handle of a TransformBox.
+    """
+    if not len(event.dims_displayed) == 2:
+        return
+
+    # we work in data space so we're axis aligned which simplifies calculation
+    # same as Layer.world_to_data
+    world_to_data = (
+        layer._transforms[1:].set_slice(event.dims_displayed).inverse
+    )
+    pos = np.array(world_to_data(event.position))[event.dims_displayed]
+
+    handle_coords = generate_transform_box_from_layer(
+        layer, event.dims_displayed
+    )
+    # TODO: dynamically set tolerance based on canvas size so it's not hard to pick small layer
+    nearby_handle = get_nearby_handle(pos, handle_coords)
+
+    # set the selected vertex of the box to the nearby_handle (can also be INSIDE or None)
+    layer._overlays['transform_box'].selected_vertex = nearby_handle
+
+
+def transform_with_box(layer, event):
```

Comment:
This function is really big. Could you split it into smaller functions to improve readability?

Reply:
Done!